webhooks/webhooks.go

```diff
@@ -0,0 +1,227 @@
+package webhooks
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+type Broadcaster interface {
+	BroadcastAction(ctx context.Context, action Action) error
+}
+
+const (
+	webhookTimeout   = 10 * time.Second
+	WebhookEventPing = "ping"
+)
+
+type (
+	Webhook struct {
+		Module string `json:"module"`
+		Event  string `json:"event"`
+		URL    string `json:"url"`
+	}
+
+	WebhookQueueInfo struct {
+		URL  string `json:"url"`
+		Size int    `json:"size"`
+	}
+
+	// Action describes an event that has been triggered.
+	Action struct {
```

Comment:
Both a `Webhook` and an `Action` have this field called `Event` now. It's a little confusing? Do we need to add the notion of an `Action`? A `ping` constitutes an event, while if I would have to guess I would've probably guessed it's an action.

Reply:
Renamed it. Didn't do so at first because it seemed strange that an `Event` would have an `event` field.